pkg/ctl/install/install.go

```diff
@@ -263,27 +264,43 @@ func (fi *fluxInstaller) addCommitAndPushFluxManifests(ctx context.Context, clon
 	return runGitCmd(pushCtx, cloneDir, "push")
 }
 
-func (fi *fluxInstaller) applyManifestsInDir(ctx context.Context, dir string) error {
-	// NOTE: invoking kubectl is not ideal, but it's the less hairy way to do it until
-	//       server-side apply is available ( https://github.com/kubernetes/enhancements/issues/555 )
-	applyCtx, applyCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
-	defer applyCtxCancel()
-	kubeconfigFile, err := ioutil.TempFile(os.TempDir(), "eksctl-flux-install-kubeconfig")
+func (fi *fluxInstaller) applyManifests(manifests map[string][]byte) error {
+	objects, err := manifestsToObjects(manifests)
 	if err != nil {
-		return fmt.Errorf("cannot find temporary kubeconfig file: %s", err)
+		return err
+	}
+	// TODO: initialise the client elsewhere so that a mock client can easily be dependency-injected for testing purposes.
+	client, err := kubeutils.NewRawClient(fi.k8sClientSet, fi.k8sRestConfig)
+	if err != nil {
+		return err
 	}
-	kubeconfigPath := kubeconfigFile.Name()
-	kubeconfigFile.Close()
-	// defer os.Remove(kubeconfigFile.Name())
-	if _, err := kubeconfig.Write(kubeconfigPath, *fi.k8sConfig, true); err != nil {
-		return fmt.Errorf("cannot write to temporary kubeconfig file (%s): %s", kubeconfigFile.Name(), err)
+	for _, object := range objects {
+		resource, err := client.NewRawResource(object)
+		if err != nil {
+			return err
+		}
+		status, err := resource.CreateOrReplace(false)
```

Comment:
Will this allow `kubectl apply`s on the resources later on? Otherwise it will break Flux syncing.

Reply:
It seems to me it does, see also this experiment: https://github.com/weaveworks/eksctl/pull/1064#discussion_r309018655
Please let me know if you'd want to test something else.